### ISSUE TITLE
Unable to parse media post

### ISSUE THREAD

**Issue by joecue** [WP-API/WP-API#2244]
I am trying to get to the URL of an image for a media post, and I'm unable to parse the Media Post using JavaScript.  I can pull the post up in my browser and I see the JSON, but when I try to retrieve the data using JavaScript I can't pull any of the parameters.

Any ideas?

**Comment by danielbachhuber** [comment]
> I can pull the post up in my browser and I see the JSON, but when I try to retrieve the data using JavaScript I can't pull any of the parameters.

Can you share more details please? Specifics, screenshots, etc.

**Comment by joecue** [comment]
Sure.. 

I have an image in my media library attached as a featured image to a post.

I can view the response in my browser to, /wp-json/wp/v2/posts/?filter[category_name]=background-photos

`[
{
id: 4,
date: "2016-02-08T16:53:15",
date_gmt: "2016-02-08T16:53:15",
guid: {
rendered: "http://lorainccc.dev/mylccc/?p=4"
},
modified: "2016-02-08T18:39:51",
modified_gmt: "2016-02-08T18:39:51",
slug: "mycampus-guest-page-background",
type: "post",
link: "http://lorainccc.dev/mylccc/2016/02/08/mycampus-guest-page-background/",
title: {
rendered: "MyCampus Guest Page Background"
},
content: {
rendered: ""
},
excerpt: {
rendered: ""
},
author: 1,
featured_media: 5,
comment_status: "open",
ping_status: "open",
sticky: false,
format: "standard",
categories: [
2
],
tags: [ ],
_links: {
self: [
{
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/posts/4"
}
],
collection: [
{
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/posts"
}
],
about: [
{
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/types/post"
}
],
author: [
{
embeddable: true,
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/users/1"
}
],
replies: [
{
embeddable: true,
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/comments?post=4"
}
],
version-history: [
{
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/posts/4/revisions"
}
],
https://api.w.org/featuredmedia: [
{
embeddable: true,
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/media/5"
}
],
https://api.w.org/attachment: [
{
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/media?parent=4"
}
],
https://api.w.org/term: [
{
taxonomy: "category",
embeddable: true,
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/categories?post=4"
},
{
taxonomy: "post_tag",
embeddable: true,
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/tags?post=4"
}
],
https://api.w.org/meta: [
{
embeddable: true,
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/posts/4/meta"
}
]
}
}
]`

I can even pull it up in my html page using the following javascript

`//Get Image Path for Background
    $.getJSON("http://lorainccc.dev/mylccc/wp-json/wp/v2/posts/?filter[category_name]=background-photos", function (data) {

```
 var photoLink = baseURL + mediaLink + data[0].featured_media;

div.append( $('<p />').append(document.createTextNode(photoLink)) ); 

 $('body').append(div);

});//End Image Path`
```

which will give me /wp-json/wp/v2/media/5

but when I go to use 

`$.getJSON("http://lorainccc.dev/mylccc/wp-json/wp/v2/media/5", function(data) {
     var div = $('<div style="border: 1px solid #9f9f9f; min-height: 15px;"/>');

```
 div.append( $('<p />') ); 

 $(div).attr('id', data[0].id));

 $('body').append(div);
});`
```

my page ends up empty unless I comment where I'm adding the ID of the post to the div id attribute, then my page will load and show my div container borders.

I can view the media url just fine in my browser

`{
id: 5,
date: "2016-02-08T16:52:53",
date_gmt: "2016-02-08T16:52:53",
guid: {
rendered: "http://lorainccc.dev/mylccc/wp-content/uploads/sites/2/2016/02/1X7A6717.jpg"
},
modified: "2016-02-08T16:52:53",
modified_gmt: "2016-02-08T16:52:53",
slug: "1x7a6717",
type: "attachment",
link: "http://lorainccc.dev/mylccc/2016/02/08/mycampus-guest-page-background/1x7a6717/",
title: {
rendered: "1X7A6717"
},
author: 1,
comment_status: "open",
ping_status: "closed",
alt_text: "",
caption: "",
description: "",
media_type: "image",
media_details: {
width: 5760,
height: 3840,
file: "2016/02/1X7A6717.jpg",
sizes: {
thumbnail: {
file: "1X7A6717-150x150.jpg",
width: 150,
height: 150,
mime_type: "image/jpeg",
source_url: "http://lorainccc.dev/mylccc/wp-content/uploads/sites/2/2016/02/1X7A6717-150x150.jpg"
},
medium: {
file: "1X7A6717-300x200.jpg",
width: 300,
height: 200,
mime_type: "image/jpeg",
source_url: "http://lorainccc.dev/mylccc/wp-content/uploads/sites/2/2016/02/1X7A6717-300x200.jpg"
},
medium_large: {
file: "1X7A6717-768x512.jpg",
width: 768,
height: 512,
mime_type: "image/jpeg",
source_url: "http://lorainccc.dev/mylccc/wp-content/uploads/sites/2/2016/02/1X7A6717-768x512.jpg"
},
large: {
file: "1X7A6717-1024x683.jpg",
width: 1024,
height: 683,
mime_type: "image/jpeg",
source_url: "http://lorainccc.dev/mylccc/wp-content/uploads/sites/2/2016/02/1X7A6717-1024x683.jpg"
},
mycampus_background: {
file: "1X7A6717-1600x1068.jpg",
width: 1600,
height: 1068,
mime_type: "image/jpeg",
source_url: "http://lorainccc.dev/mylccc/wp-content/uploads/sites/2/2016/02/1X7A6717-1600x1068.jpg"
},
post-thumbnail: {
file: "1X7A6717-1200x800.jpg",
width: 1200,
height: 800,
mime_type: "image/jpeg",
source_url: "http://lorainccc.dev/mylccc/wp-content/uploads/sites/2/2016/02/1X7A6717-1200x800.jpg"
},
full: {
file: "1X7A6717.jpg",
width: 5760,
height: 3840,
mime_type: "image/jpeg",
source_url: "http://lorainccc.dev/mylccc/wp-content/uploads/sites/2/2016/02/1X7A6717.jpg"
}
},
image_meta: {
aperture: "9",
credit: "",
camera: "Canon EOS 5D Mark III",
caption: "",
created_timestamp: "1412705357",
copyright: "",
focal_length: "35",
iso: "640",
shutter_speed: "0.005",
title: "",
orientation: "0",
keywords: [ ]
}
},
post: 4,
source_url: "http://lorainccc.dev/mylccc/wp-content/uploads/sites/2/2016/02/1X7A6717.jpg",
_links: {
self: [
{
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/media/5"
}
],
collection: [
{
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/media"
}
],
about: [
{
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/types/attachment"
}
],
author: [
{
embeddable: true,
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/users/1"
}
],
replies: [
{
embeddable: true,
href: "http://lorainccc.dev/mylccc/wp-json/wp/v2/comments?post=5"
}
]
}
}`

So I'm just kinda stumped as to why a regular post works in JavaScript and the browser, but the media post only works in the browser.

Basically I'm trying to get after the featured image URL, so a non tech admin can change the image, and by using the API on our Student Information System portal via JavaScript, load a different image into the background of the portal guest page.

Maybe my JavaScript is incorrect, but it's been working thus far.  I'm currently pulling a list of announcements onto the guest page via the API and it's works fantastic.

Any help you can provide would be appreciated!

**Comment by danielbachhuber** [comment]
This:

```
$.getJSON("http://lorainccc.dev/mylccc/wp-json/wp/v2/media/5", function(data) {
$(div).attr('id', data[0].id));
```

Should be this:

```
$.getJSON("http://lorainccc.dev/mylccc/wp-json/wp/v2/media/5", function(data) {
$(div).attr('id', data.id));
```

When you're accessing an individual resource, not a collection, you're getting an object in response, which means you don't need the `[0]`

**Comment by joecue** [comment]
I've tried that as well, and still would error.  I'll give it another go.

I put my code into a gist, it's posted here, https://gist.github.com/joecue/e2f232825a605c1b2a2a

## 

UPDATE: the line to pull the ID had an extra ')' in it.  I updated the gist, as the code now works.
